backend/src/main/java/com/woowacourse/gongcheck/application/GithubOauthClient.java

```diff
@@ -0,0 +1,68 @@
+package com.woowacourse.gongcheck.application;
+
+import com.woowacourse.gongcheck.application.response.GithubAccessTokenResponse;
+import com.woowacourse.gongcheck.application.response.GithubProfileResponse;
+import com.woowacourse.gongcheck.exception.UnauthorizedException;
+import com.woowacourse.gongcheck.presentation.request.GithubAccessTokenRequest;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class GithubOauthClient {
+
+    private static final String ACCEPT_HEADER = "Accept";
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private static final String BEARER_TYPE = "Bearer ";
+
+    private final String clientId;
+    private final String clientSecret;
+    private final String tokenUrl;
+    private final String profileUrl;
+    private final RestTemplate restTemplate;
+
+    public GithubOauthClient(@Value("${github.client.id}") final String clientId,
+                             @Value("${github.client.secret}") final String clientSecret,
+                             @Value("${github.url.token}") final String tokenUrl,
+                             @Value("${github.url.profile}") final String profileUrl,
+                             final RestTemplate restTemplate) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.tokenUrl = tokenUrl;
+        this.profileUrl = profileUrl;
+        this.restTemplate = restTemplate;
+    }
+
+    public String requestAccessToken(String code) {
+        GithubAccessTokenRequest githubAccessTokenRequest = new GithubAccessTokenRequest(code, clientId, clientSecret);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(ACCEPT_HEADER, MediaType.APPLICATION_JSON_VALUE);
+        HttpEntity<MultiValueMap<String, String>> httpEntity = new HttpEntity(githubAccessTokenRequest, headers);
+
+        String accessToken = restTemplate
+                .exchange(tokenUrl, HttpMethod.POST, httpEntity, GithubAccessTokenResponse.class)
+                .getBody()
+                .getAccessToken();
+        if (accessToken == null) {
```

Comment:
```suggestion
        if (Objects.isNull(accessToken)) {
```
최대한 제공되는 메소드를 사용함으로써 메소드 네이밍에 의미를 담아주는것이 중요합니다 :)

Reply:
😀👍